### DISCUSSION TITLE
Request for clarification of BPP API functions and callbacks

### DISCUSSION THREAD

**Discussion by sudhirrg** [beckn/mobility#38] [Q&A] [answered]
Hi all, my team is implementing a BPP including the API functions together with the respective callbacks.
We have a few queries about the message formats and how to be compliant with them.
We intend to comply with the Beckn Mobility API v1.0.0
1. - The version 1.0.0 for search --->fulfillment-->intent--->stops[]: this just tells the different gps cordinates involved in fulfilment. However it does not say what type of location cordinates such as start or end.How do we distinguish between the start location and the end location?

**Comment by ravi-prakash-v** [comment, marked as answer]
Hi @sudhirrg. Following are the responses to your queries

> The version 1.0.0 for search --->fulfillment-->intent--->stops[]: this just tells the different gps cordinates involved in fulfilment. However it does not say what type of location cordinates such as start or end.How do we distinguish between the start location and the end location?

The `Stop` schema has a property called `type`. For mobility, the type could take the following values

- START
- END

There could be more depending upon the nature of the use case. These enuerations have not been specified yet in the mobility specification. Kindly bear with us till we update the spec. But for now, you can use `fulfillment.stops[].type = START` and `fulfillment.stops[].type = END` to indicate the start and end of a ride. 

Regarding your other questions, can I request you to edit this post and ask each question as a separate discussion thread? Thanks

**Reply by sudhirrg** [reply]
We will use **fulfillment.stops[].type** to indicate the starting and ending points of a ride, as you suggest.
I entered our other questions in separate threads.

**Comment by yashb042** [comment]
@sudhirrg  Could you please assist me in registering with the beckn protocol, can't seem to find the URL/documentation supporting the same.

**Reply by sudhirrg** [reply]
Yash, are you trying to register your BPP implementation (or BAP implementation) with the Beckn registry? Please provide some details to clarify your question.

**Reply by sudhirrg** [reply]
@yashb042 If you wish to register on the open Beckn network, you will need to register as a Beckn participant at registry.becknprotocol.io. You need to add entries and corresponding keys in the Beckn registry.
